libs/sdk-core/src/persist/transactions.rs

```diff
@@ -345,6 +393,21 @@ fn filter_to_where_clause(
         }
     }
 
+    if let Some(filters) = metadata_filter {
+        if let Ok(map) = serde_json::from_str::<Map<String, Value>>(&filters) {
+            map.iter().for_each(|(key, value)| {
+                where_clause.push(format!(
+                    "metadata->'$.{}' = '{}'",
+                    key,
+                    match value {
+                        Value::String(s) => format!(r#""{s}""#),
+                        _ => value.to_string(),
+                    }
+                ))
```

Comment:
Thanks @yaslama for the reply. This does indeed simplify even more our approach, although I noticed we still need to wrap JSON strings inside some double quotes in order to make this work as intended. Still, a huge improvement over checking for null and boolean fields, as now it can be done with only one query 👍

Reply:
> although I noticed we still need to wrap JSON strings inside some double quotes in order to make this work as intended

I don't understand this sentence. In order to be a valid json, a string has to be enclosed by double quotes. So if the second parameter (the value of the filter) is required to be a json, why it's a problem?

Reply:
Since we need to parse the filter string (a JSON object) and split it into multiple where-clauses for each field, we need to iterate over a `Map<String, Value>`. When a field has a `Value::String`, and we call the method `to_string()` (used in `format!` macro), the string gets printed out without the double quotes, which is a problem for the SQL query as it only works as follows: `SELECT * FROM metadata->'$.{fieldName}' = '"{value}"'`

Reply:
@hydra-yse I think What Akiba's suggestion is how to solve the issue of searching in non string values.
And by solving this it allows us to go back to the previously suggested interface for filtering instead of passing a json string as filter.

We can pass a list of `PaymentMetadataFilter` where the struct is defined like this:
```
struct PaymentMetadataFilter {
  search_path: String,
  search_value: String
}
```

Then it is trivial to compose the where clause using the arrow operator and we get two benefits:
1. The interface is clear (we pass a list of filters with documented fields)
2. It will search in all values (including booleans, numbers)

What do you think?

Reply:
Sure, they would both work well. On one hand I like the simplicity of being able to pass a JSON string without requiring any additional data structure, but I also see how setting the filters this way would make it very similar (and thus more familiar) to the `type_filters`. I'll continue as you suggested